Jenkinsfile

```
@@ -1,92 +1,155 @@
+    ),
+    string(
+      name: 'GCC_VERSION',
+      defaultValue: '11',
+      description: 'Version of gcc to load',
+      trim: true
+    )
+  ])
+])
+
 def get_agent(String jobname) {
   if (jobname.contains('linux')) {
-    withCredentials([string(credentialsId: 'manylinux_agent', variable: 'agent')]) {
-      return "${agent}"
-    }
+    return "rocky8"
   } else if (jobname.contains('windows')) {
-    withCredentials([string(credentialsId: 'win10_agent', variable: 'agent')]) {
-      return "${agent}"
-    }
+
+    return "icdpacewin"
+
   } else {
     return ''
   }
 }
 
 def get_github_token() {
-  withCredentials([string(credentialsId: 'pace_python_release', variable: 'github_token')]) {
+  withCredentials([string(credentialsId: 'GitHub_API_Token', variable: 'github_token')]) {
     return "${github_token}"
   }
 }
 
-def setGitHubBuildStatus(String status, String message) {
-    script {
-        withCredentials([string(credentialsId: 'PacePython_API_Token',
-                variable: 'api_token')]) {
-          if (isUnix()) {
-            sh """
-                curl -H "Authorization: token ${api_token}" \
-                --request POST \
-                --data '{ \
-                    "state": "${status}", \
-                    "description": "${message} on ${env.JOB_BASE_NAME}", \
-                    "target_url": "$BUILD_URL", \
-                    "context": "${env.JOB_BASE_NAME}" \
-                }' \
-                https://api.github.com/repos/pace-neutrons/pace-python/statuses/${env.GIT_COMMIT}
-            """
-          }
-          else {
-            return powershell(
-            script: """
-                \$body = @"
-                  {
-                    "state": "${status}",
-                    "description": "${message} on ${env.JOB_BASE_NAME}",
-                    "target_url": "$BUILD_URL",
-                    "context": "${env.JOB_BASE_NAME}"
-                  }
-"@
-                [Net.ServicePointManager]::SecurityProtocol = "tls12, tls11, tls"
-                Invoke-RestMethod -URI "https://api.github.com/repos/pace-neutrons/pace-python/statuses/${env.GIT_COMMIT}" \
-                    -Headers @{Authorization = "token ${api_token}"} \
-                    -Method 'POST' \
-                    -ContentType "application/json" \
-                    -Body \$body
-            """,
-            returnStdout: true
-            )
-          }
-        }
-    }
+def name_conda_env(String python_version) {
+  def env_name = "py" + python_version.replace(".","")
+  return env_name
 }
 
+
 pipeline {
 
   agent {
     label get_agent(env.JOB_BASE_NAME)
   }
 
+  environment {
+    ENV_NAME = name_conda_env(env.PYTHON_VERSION)
+  }
+
   stages {
 
+    stage('Notify') {
+      steps {
+        post_github_status("pending", "The build is running")
+      }
+    }
+
+    stage('Display-Environment-Variables') {
+      steps {
+        script {
+          if (isUnix()) {
+            sh 'env'
```

Strictly the command here is:
```suggestion
            sh 'printenv'
```
But bare `env` will do the same job on most implementations